### PR TITLE
Fix for #847. Implemented findSubtypes.

### PR DESCRIPTION
**Pull request by MichaelDoumin**
Since the method findSubtypes was not implemented after jackson-swagger-module had been moved to the  swagger-core, ModelConverter could not find subtypes of passed class. Fixed it. Added unit-test.

### CODE REVIEW

**Comment by webron** [comment]
Related to #847.

**Comment by tomtit** [comment]
Looks like this snippet:
```
+    final JsonSubTypes jst = a.getAnnotation(JsonSubTypes.class);
+    if (jst != null) {
+      final JsonSubTypes.Type[] subTypes = jst.value();
+      final List<NamedType> names = new ArrayList<NamedType>(subTypes.length);
+      for (JsonSubTypes.Type subType : subTypes) {
+        names.add(new NamedType(subType.value()));
+      }
+      if (!names.isEmpty()) {
+        return names;
+      }
+ }
```
duplicates FasterXML internals.
Deep inside we have instance of the `com.fasterxml.jackson.databind.introspect.AnnotationIntrospectorPair` class:
```
    @Override
    public List<NamedType> findSubtypes(Annotated a)
    {
        List<NamedType> types1 = _primary.findSubtypes( a );
        List<NamedType> types2 = _secondary.findSubtypes( a );
        if (types1 == null || types1.isEmpty()) return types2;
        if (types2 == null || types2.isEmpty()) return types1;
        ArrayList<NamedType> result = new ArrayList<NamedType>(types1.size() + types2.size());
        result.addAll(types1);
        result.addAll(types2);
        return result;
    }
```
where `_primary` holds our SwaggerAnnotationIntrospector and `_secondary` holds some internal class that handles JSON annotations.
At least as we can see from our [CompositionTest.scala](https://github.com/swagger-api/swagger-core/blob/develop_2.0/modules/swagger-core/src/test/scala/CompositionTest.scala) JSON annotations are already supported.

**Comment by MichaelDoumin** [comment]
The last fix considers Alexey's remark above: JsonSubTypes processing is re-used from JacksonAnnotationIntrospector.

File: modules/swagger-core/src/main/java/com/wordnik/swagger/jackson/AbstractModelConverter.java [hunk 1]
```diff
@@ -35,7 +37,10 @@ protected AbstractModelConverter(ObjectMapper mapper) {
             new SimpleModule("swagger", Version.unknownVersion()) {
               @Override
               public void setupModule(SetupContext context) {
```
Comment (tomtit) [inline modules/swagger-core/src/main/java/com/wordnik/swagger/jackson/AbstractModelConverter.java]: I think this change is not required at all.

File: modules/swagger-core/src/main/java/com/wordnik/swagger/jackson/AbstractModelConverter.java [hunk 2]
```diff
@@ -35,7 +37,10 @@ protected AbstractModelConverter(ObjectMapper mapper) {
...
-                context.insertAnnotationIntrospector(new SwaggerAnnotationIntrospector());
+                context.insertAnnotationIntrospector(
+                        new AnnotationIntrospectorPair(
+                                new SwaggerAnnotationIntrospector(),
+                                new JacksonAnnotationIntrospector()));
```
Comment (tomtit) [inline modules/swagger-core/src/main/java/com/wordnik/swagger/jackson/AbstractModelConverter.java]: I think this change is not required at all.

Comment (fehguy) [inline modules/swagger-core/src/main/java/com/wordnik/swagger/jackson/AbstractModelConverter.java]: I agree, I believe that the JacksonAnnotatinIntrospector is embedded in SwaggerAnnotationIntrospector

**Pull request merged**